compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/symbols/BInvokableTypeSymbol.java

```diff
@@ -42,15 +43,14 @@
     public BVarSymbol restParam;
     public BType returnType;
     public List<BLangAnnotationAttachment> returnTypeAnnots;
-    public Map<String, BType> paramDefaultValTypes;
-
+    public Map<String, BLangLambdaFunction> defaultValues;
```

Comment:
Why are we keeping the lambda function itself in the type symbol? It should be just symbols?

Reply:
To keep a reference to the closure of the default value.

Reply:
IMO, this is not correct, we should be referring to symbols (and not nodes/expressions) from a symbol. 

Looping in @hasithaa also.

Reply:
And due to this, seems like `BLangLambdaFunction`s and `BLangFunction`s are also being created at BIR symbol enter, which again doesn't seem correct to me.

https://github.com/ballerina-platform/ballerina-lang/blob/83d6990ba1d6cec35d4f20ecb1337ec93b8fd035/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java#L1044

Reply:
Fixed.